src/monster.h

```diff
@@ -5,6 +5,7 @@
 #include "creature.h"
 #include "enums.h"
 #include "int_id.h"
+#include "map.h"
```

Comment:
You should forward declare `map_stack` instead of included the whole "map.h" file.